tests/DataTransferTransferObjectTest.php

```diff
@@ -248,4 +248,40 @@ public function nested_dtos_are_automatically_cast_from_arrays_to_objects()
         $this->assertEquals('parent', $object->name);
         $this->assertEquals('child', $object->child->name);
     }
+
+    /** @test */
+    public function nested_dtos_are_recursive_cast_from_object_to_array_when_to_array()
```

Comment:
Add a separate test for every possible path, instead of combining three cases into one.

Reply:
ok